static/app/components/sidebar/sidebarAccordion.tsx

```diff
@@ -14,15 +27,44 @@ type SidebarAccordionProps = SidebarItemProps & {
   children?: React.ReactNode;
 };
 
+export const ExpandedContext = createContext<{
+  openMainItemId: string | null;
+  setOpenMainItem: (mainItemId: string | null) => void;
+}>({
+  openMainItemId: null,
+  setOpenMainItem: () => {},
+});
+
+export function ExpandedContextProvider(props) {
+  const [openMainItemId, setOpenMainItem] = useState<string | null>(null);
+
+  return (
+    <ExpandedContext.Provider value={{openMainItemId, setOpenMainItem}}>
+      {props.children}
+    </ExpandedContext.Provider>
+  );
+}
+
```

Comment:
This should be in its own file, I think. I got confused because it was declared here, but exported to the main `Sidebar`, and I thought this was wrapped around _this_ component even though it's not